### PR TITLE
Split package into tools and lib subpackages

### PR DESCRIPTION
**Pull request by pavelzw**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,31 +1,142 @@
...
+        - {{ compiler('c') }}
+        - {{ stdlib('c') }}
+    test:
+      commands:
+        - test ! -f ${PREFIX}/bin/dumpsexp
+        - test ! -f ${PREFIX}/bin/hmac256
+        - test ! -f ${PREFIX}/bin/mpicalc
+        - test -f ${PREFIX}/lib/libgcrypt.20${SHLIB_EXT}  # [osx]
+        - test -f ${PREFIX}/lib/libgcrypt${SHLIB_EXT}.20  # [linux]
+        - test -f ${PREFIX}/lib/libgcrypt${SHLIB_EXT}.20.5.0  # [linux]    about:
```
Comment (xhochy) [inline recipe/meta.yaml:102]:
```suggestion
        - test -f ${PREFIX}/lib/libgcrypt${SHLIB_EXT}.20.5.0  # [linux]
     about:
```

File: recipe/install-libgcrypt-devel.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+make install
+rm $PREFIX/bin/dumpsexp
+rm $PREFIX/bin/hmac256
+rm $PREFIX/bin/mpicalc
+rm $PREFIX/bin/libgcrypt-config
```
Comment (xhochy) [inline recipe/install-libgcrypt-devel.sh:9]: This is required for development
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by pavelzw** [comment]
@conda-forge/libgcrypt this is ready for review

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
@pavelzw Please also update the build of `libsecret` to use the new output. Happy to review the PR over there.

**Pull request merged**